trie/trie.go

```diff
@@ -266,15 +270,19 @@ func (t *Trie) tryGetWithEpoch(origNode node, key []byte, pos int, epoch types.S
 		}
 
 		if updateEpoch {
-			n.setEpoch(t.currentEpoch)
-			n.UpdateChildEpoch(int(key[pos]), t.currentEpoch)
 			value, newnode, didResolve, err = t.tryGetWithEpoch(n.Children[key[pos]], key, pos+1, t.currentEpoch, true)
 		} else {
 			value, newnode, didResolve, err = t.tryGetWithEpoch(n.Children[key[pos]], key, pos+1, n.GetChildEpoch(int(key[pos])), false)
 		}
-		if err == nil && didResolve {
-			n = n.copy()
-			n.Children[key[pos]] = newnode
+		if err == nil {
+			if didResolve || (updateEpoch && epoch != t.currentEpoch) {
+				n = n.copy()
+				n.Children[key[pos]] = newnode
+				if updateEpoch && epoch != t.currentEpoch {
+					n.setEpoch(t.currentEpoch)
+					n.UpdateChildEpoch(int(key[pos]),  t.currentEpoch)
```

Comment:
when child is nil, may need not update its epoch?

Reply:
if child is nil, err could not be nil